### PR TITLE
Documentation updates for 3.2

### PR DESCRIPTION
**Pull request by evildmp**
Includes:

* release notes
* updated user tutorial
* new features
* changes in line with new touchscreen interfaces
* various tidying up

### CODE REVIEW

File: docs/upgrade/3.2.rst [hunk 1]
```diff
@@ -14,10 +14,20 @@ possible.
     from a previous version. Please read :ref:`upgrade-guide-3.2` for a step-by-step guide to the
     process of upgrading from 3.1 to 3.2.
 
+
 *****************
 What's new in 3.2
 *****************
 
+* full touch-screen support at all sizes from mobile phone to table-top device
```
Comment (FinalAngel) [inline docs/upgrade/3.2.rst]: I'd just mention tablets / phablets

File: docs/upgrade/3.2.rst [hunk 2]
```diff
@@ -14,10 +14,20 @@ possible.
...
+* enhanced and polished user interface
+* much-needed improvements to the structure-board
+* enhancements to components such as the pop-up plugin editor, sideframe and the toolbar
+* significant speed improvements on loading, HTTP requests and file sizes
+* new *Link/Button* look & feel
```
Comment (FinalAngel) [inline docs/upgrade/3.2.rst]: what does that refer to ?

File: docs/upgrade/3.2.rst [hunk 3]
```diff
@@ -26,12 +36,14 @@ Renaming cms_app, cms_toolbar, menu modules
 
 Old names are still supported but deprecated; support will be removed in 3.4.
 
+
 Action required
 ---------------
 
 In your own applications that use these modules, rename ``cms_app.py`` to ``cms_apps.py``,
 ``cms_toolbar.py`` to ``cms_toolbar.py`` and ``menu.py`` to ``menus.py``.
```
Comment (FinalAngel) [inline docs/upgrade/3.2.rst]: to ``cms_toolbars.py`` ? did that change @mkoistinen

File: docs/upgrade/3.2.rst [hunk 4]
```diff
@@ -42,6 +54,20 @@ other changes in the CMS for version 3.2 provides automatic reloading of URLs
...
+===============
+
+SASS-related changes
+--------------------
+
+We now use LibSass rather than Compass for building static files (this only affects frontend
+developers *of* django CMS - contributors to it, not other users or developers). We've also adopted
+CSSComb.
+
+.. todo:: link to the documentation for this
```
Comment (FinalAngel) [inline docs/upgrade/3.2.rst]: docs are here: https://github.com/divio/django-cms/blob/develop/docs/contributing/code.rst#frontend

**Comment by yakky** [comment]
@evildmp ping me when ready for merge
And add [ci skip] to the commit description to avoid multiple CI runs, I will check the correct doc format before merging

File: docs/upgrade/3.2.rst [hunk 5]
```diff
@@ -26,29 +42,60 @@ Renaming cms_app, cms_toolbar, menu modules
...
+
+SASS-related changes
+--------------------
 
-*****************************
-Backward incompatible changes
-*****************************
+We now use `LibSass <https://github.com/sass/libsass>`_ rather than Compass for building static
+files (this only affects frontend developers *of* django CMS - contributors to it, not other users
+or developers). We've also adopted `CSSComb <http://csscomb.com>`_.
```
Comment (vxsx) [inline docs/upgrade/3.2.rst]: we adopted it, but didn't actually include `.csscomb.json` to the repo, it's coming from https://github.com/aldryn/aldryn-boilerplate-bootstrap3/blob/master/.csscomb.json

File: docs/upgrade/3.2.rst (hunk 3 above)
Comment (mkoistinen) [inline docs/upgrade/3.2.rst]: Yes.

**Comment by mkoistinen** [comment]
@evildmp can you rebase this? The errors you're seeing have been fixed long ago.

File: cms/static/cms/sass/components/_toolbar.scss [hunk 6]
```diff
@@ -112,9 +112,11 @@
         ul {
             display: block;
             position: absolute;
+            overflow: auto;
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_toolbar.scss]: weird that this is in here ?_?

File: docs/how_to/apphooks.rst [hunk 7]
```diff
@@ -36,6 +36,8 @@ a page and open the advanced settings tab. Select your new apphook under
     an apphook or the slug if a page which has a descendant with an apphook, you
     have to restart your server to re-load the URL caches.
 
+    If you have the
```
Comment (FinalAngel) [inline docs/how_to/apphooks.rst]: whats this?

File: docs/upgrade/3.2.rst [hunk 8]
```diff
@@ -14,10 +14,45 @@ possible.
...
 
+* :ref:`new welcome page <create-first-page>` to help new users
+* full touch-screen support at all sizes from small tablets to table-top devices
+* enhanced and polished user interface
+* much-needed improvements to the structure-board
+* enhancements to components such as the pop-up plugin editor, sideframe and the toolbar
+* significant speed improvements on loading, HTTP requests and file sizes
+* restarts are no longer required when changing apphook configurations
+* a new content wizard system, adaptable to arbitrary content types
+* new *Link/Button* look & feel
```
Comment (FinalAngel) [inline docs/upgrade/3.2.rst]: whats that?

File: docs/upgrade/3.2.rst (hunk 3 above)
Comment (FinalAngel) [inline docs/upgrade/3.2.rst]: typo ``cms_toolbar.py`` to ``cms_toolbars.py``

File: docs/upgrade/3.2.rst [hunk 9]
```diff
@@ -26,40 +61,117 @@ Renaming cms_app, cms_toolbar, menu modules
...
-* Fix placeholder template tags ignoring the ``lang`` parameter
+
+New structure board
+===================
+
+The structure board is cleaner and easier to understand. It now displays its elements in a tree,
+rather than in a series of nested boxes.
+
+You can optionally enable the old appearance and behaviour with the
+:ref:`CMS_TOOLBAR_SIMPLE_STRUCTURE_MODE` setting.
```
Comment (FinalAngel) [inline docs/upgrade/3.2.rst]: this will be removed in 3.3

File: docs/how_to/apphooks.rst (hunk 7 above)
Comment (evildmp) [inline docs/how_to/apphooks.rst]: Fixed.

**Comment by FinalAngel** [comment]
LGTM

**Pull request merged**